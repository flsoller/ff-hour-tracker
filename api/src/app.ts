// Package imports
import express, { Application } from 'express';
import cors from 'cors';
import dotenv from 'dotenv';

// Middleware imports
import errorHandler from './middleware/errorHandler';

// Controller imports
import organizations from './controllers/Organizations';
import members from './controllers/Members';
import activities from './controllers/Activities';
import timelog from './controllers/TimeLogs';

// Load env configuration
dotenv.config();

// Initalize express
const app: Application = express();

// Allow cors for development
if (process.env.NODE_ENV === 'development') {
  app.use(cors());
}

// JSON body-parser middleware
app.use(express.json());

// Mount routes
const API_V0 = '/api/v0';
app.use(`${API_V0}/organizations`, organizations);
app.use(`${API_V0}/members`, members);
app.use(`${API_V0}/activities`, activities);
app.use(`${API_V0}/timelog`, timelog);

// Error handling middleware
app.use(errorHandler);

export default app;
