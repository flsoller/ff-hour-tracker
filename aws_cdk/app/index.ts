#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import { HourTrackerApi } from '../lib';

const app = new cdk.App();
new HourTrackerApi(app, 'HourTrackerApi');

app.synth();
