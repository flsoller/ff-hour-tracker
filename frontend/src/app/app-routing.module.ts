import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
  { path: '', redirectTo: '/dashboard', pathMatch: 'full' },
  {
    path: 'dashboard',
    loadChildren: () =>
      import('./dashboard/dashboard.module').then(
        (module) => module.DashboardModule
      ),
  },
  {
    path: 'config',
    loadChildren: () =>
      import('./configuration/configuration.module').then(
        (module) => module.ConfigurationModule
      ),
  },
  {
    path: 'reports',
    loadChildren: () =>
      import('./reports/reports.module').then((module) => module.ReportsModule),
  },
  {
    path: 'timelog',
    loadChildren: () =>
      import('./time-logging/time-logging.module').then(
        (module) => module.TimeLoggingModule
      ),
  },
  {
    path: 'users',
    loadChildren: () =>
      import('./user-management/user-management.module').then(
        (module) => module.UserManagementModule
      ),
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
