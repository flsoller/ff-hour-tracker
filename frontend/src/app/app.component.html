<div class="app-container">
  <app-sidebar></app-sidebar>
  <router-outlet></router-outlet>
</div>
