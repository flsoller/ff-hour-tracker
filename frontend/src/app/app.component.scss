.app-container {
  height: 100%;
  display: flex;
}
