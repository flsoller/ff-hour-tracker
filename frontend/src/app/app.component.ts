import { HttpClient } from '@angular/common/http';
import { Component, OnInit } from '@angular/core';
import { environment } from 'src/environments/environment';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  title = 'frontend';
  apiRes = '';

  apiURL = environment.apiURL;

  constructor(private http: HttpClient) {}

  ngOnInit() {}
}
