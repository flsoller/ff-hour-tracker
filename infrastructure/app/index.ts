#!/usr/bin/env node
import "source-map-support/register";
import * as cdk from "aws-cdk-lib";
import { HourTrackerApi } from "../lib/api";
import { HourTrackerImageRepositories } from "../lib/ecr";
import { HourTrackerOrganizationManager } from "../lib/org-manager";

const app = new cdk.App();
new HourTrackerImageRepositories(app, "HourTrackerImageRepositories");
new HourTrackerApi(app, "HourTrackerApi");
new HourTrackerOrganizationManager(app, "HourTrackerOrganizationManager");

app.synth();
