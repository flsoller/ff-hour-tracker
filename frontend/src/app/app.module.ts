import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { HttpClientModule } from '@angular/common/http';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { NavigationModule } from './navigation/navigation.module';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { SharedModule } from './shared/shared.module';

@NgModule({
  declarations: [AppComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    NavigationModule,
    FontAwesomeModule,
    SharedModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
