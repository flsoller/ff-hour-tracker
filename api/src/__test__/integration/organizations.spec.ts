import app from '../../app';
import supertest from 'supertest';
import {
  createUserForOrganization,
  loginTestUser,
  createOrganizations,
} from '../helpers/helpers';
import { Role, User } from '@prisma/client';

describe('createOrganization', () => {
  const ROUTE = '/api/v0/organizations';
  let token: string;
  let testUser: User;
  let adminUser: User;
  let adminUserToken: string;

  beforeEach(async () => {
    const [organization] = await createOrganizations();
    testUser = await createUserForOrganization(organization.id);
    adminUser = await createUserForOrganization(organization.id, {
      emailAddress: 'admin@user.com',
      role: Role.ADMIN,
    });
    token = loginTestUser(testUser.id);
    adminUserToken = loginTestUser(adminUser.id);
  });

  it('should create an organization', async () => {
    const res = await supertest(app)
      .post(ROUTE)
      .auth(adminUserToken, { type: 'bearer' })
      .send({
        name: 'TestOrganiztion',
        description: 'I am a happy description',
      });

    expect(res.status).toBe(201);
    expect(res.body).toMatchSnapshot({
      id: expect.any(String),
    });
  });

  it('should fail when required fields are missing', async () => {
    const res = await supertest(app)
      .post(ROUTE)
      .auth(adminUserToken, { type: 'bearer' })
      .send({
        description: 'I am a happy description',
      });

    expect(res.status).toBe(400);
    expect(res.body).toStrictEqual({
      error: 'MissingInformation',
      additionalInfo: {},
    });
  });

  it('should fail when not authenticated', async () => {
    const res = await supertest(app)
      .post(ROUTE)
      .auth('faketoken', { type: 'bearer' })
      .send({
        name: 'TestOrganiztion',
        description: 'I am a happy description',
      });

    expect(res.status).toBe(401);
    expect(res.body).toStrictEqual({
      error: 'NotAuthorized',
      additionalInfo: {},
    });
  });

  it('should only allow admin users to create a new organization', async () => {
    const res = await supertest(app)
      .post(ROUTE)
      .auth(token, { type: 'bearer' })
      .send({
        name: 'TestOrganiztion',
        description: 'I am a happy description',
      });

    expect(res.status).toBe(403);
    expect(res.body).toStrictEqual({
      error: 'NotAuthorized',
      additionalInfo: {},
    });
  });
});
