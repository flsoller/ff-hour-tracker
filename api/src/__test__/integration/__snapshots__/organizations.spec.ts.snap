// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createOrganization should create an organization 1`] = `
Object {
  "id": Any<String>,
  "name": "TestOrganiztion",
}
`;
